libs/mock-builder/src/lib.rs

```diff
@@ -11,10 +11,139 @@
+//! `mock-builder` allows you to create *mock pallets*.
+//! A *mock pallet* is a regular pallet that implements some traits whose
+//! behavior can be implemented on the fly by closures. They are perfect for
+//! testing because they allow you to customize each test case, getting
+//! organized and accurate tests for your pallet. *Mock pallet* is not just a
+//! trait mocked, it's a whole pallet that can implement one or more traits and
+//! can be added to runtimes.
 //!
-//! [`register_call!()`] and [`execute_call!()`] expect the following storage in
-//! the pallet. It's safe to just copy and paste in your pallet mock.
+//! # Motivation
+//!
+//! Pallets have dependencies. Programming in a
+//! [loosely coupled](https://docs.substrate.io/build/pallet-coupling)
+//! way is great for getting rid of those dependencies for the implementation.
+//! Nevertheless, those dependencies still exist in testing because when the
+//! `mock.rs` file is defined, you're forced to give some implementations for
+//! the associated types of your pallet `Config`.
+//!
+//! Then, you are mostly forced to use other pallet configurations
+//! geting a [tight coupling](https://docs.substrate.io/build/pallet-coupling/)
+//! with them. It has some downsides:
+//! - You need to learn how to configure other pallets.
+//! - You need to know how those pallets work, because they affect directly the
+//!   behavior of the pallet you're testing.
+//! - The way they work can give you non-completed tests. It means that some
+//!   paths of your pallet can no be tested because some dependency works in a
+//!   specific way.
+//! - You need a lot of effort maintaining your tests because each time one
+//!   dependency changes, it can easily break your tests.
+//!
+//! This doesn't scale well. Frequently some pallet dependencies need in turn to
+//! configure their own dependent pallets, making this problem even worst.
+//!
+//! This is why mocking is so important. It lets you get rid of all these
+//! dependencies and related issues, obtaining **loose coupling tests**.
+//!
+//! There are other crates focusing on this problem,
+//! such as [`mockall`](https://docs.rs/mockall/latest/mockall/),
+//! but they mock traits. Instead, this crate gives you an entire pallet
+//! ready to use in any runtime, implementing the number of traits you specify.
+//!
+//! ## *Mock pallet* usage
+//!
+//! Suppose that in our pallet, which we'll call it `my_pallet`, we have an
+//! associated type in our `Config`, which implements a `TraitA` and `TraitB`
+//! traits. Those traits are defined as follows:
+//!
+//! ```no_run
+//! trait TraitA {
+//!     type AssocA;
+//!
+//!     fn foo() -> Self::AssocA;
+//! }
+//!
+//! trait TraitB {
+//!     type AssocB;
+//!
+//!     fn bar(a: u64, b: Self::AssocB) -> u32;
+//! }
+//! ```
+//!
+//! We have a real huge pallet that implements a specific behavior for those
+//! traits, but we want to get rid of such dependency so we [generate a *mock
```

Comment:
```suggestion
//! traits, but we want to get rid of such dependency so we generate a [*mock
```

Reply:
I wanted to pick up the `generate` part because I'm redirecting to the creation section